testdata/sqllogictests/special_characters.slt

```diff
@@ -0,0 +1,9 @@
+# https://github.com/GlareDB/glaredb/issues/2309
+statement ok
+select ';';
+
+statement ok
+select ';"';
+
+statement error Schema error: No field named ";".
+select ";";
```

Comment:
Since these are going of the postgres protocol, these queries should already be working correctly since there's no checks for ';' in that code path. 

It's likely we'll want a cli test for checking that changes in this pr fix https://github.com/GlareDB/glaredb/issues/2309.

Reply:
I added a bunch of unit tests [here](https://github.com/GlareDB/glaredb/pull/2307/files#diff-99e5c07101a26683f4a227db867a7280be5cbbe8426bd9ed377ca8aa88e45cb3R304) that test the completion logic. 

I'll also add one or two integration tests just to be safe.

Reply:
> Since these are going of the postgres protocol, 

also keep in mind, all of these tests are also ran over the flightsql protocol.